storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1570,4 +1574,49 @@ public static boolean isLocalhostAddress(String address) {
         }
         return result;
     }
+
+    private static class JarConfigReader {
+        private Yaml yaml;
+        private Map<String, Object> defaultsConf;
+        private Map<String, Object> stormConf;
+        private File f;
+
+        public JarConfigReader(Yaml yaml, Map<String, Object> defaultsConf, Map<String, Object> stormConf, File f) {
+            this.yaml = yaml;
+            this.defaultsConf = defaultsConf;
+            this.stormConf = stormConf;
+            this.f = f;
+        }
+
+        public Map<String, Object> getDefaultsConf() {
+            return defaultsConf;
+        }
+
+        public Map<String, Object> getStormConf() {
+            return stormConf;
+        }
+
+        public JarConfigReader invoke() throws IOException {
```

Comment:
Nit: This method seems to be part of initializing this class. Consider making it private and calling it from the constructor.

Reply:
The class is made from IntelliJ's extract feature. We are now calling readZip/readJar separately so hopefully that is OK to keep it public and be called from outside.

Reply:
Looks good, thanks. Not sure if it makes a difference to open the jarfile with JarFile rather than ZipFile, but it can't hurt.